prow/cmd/pipeline/controller.go

```diff
@@ -317,6 +331,18 @@ func (c *controller) deletePipelineRun(pContext, namespace, name string) error {
 	return p.client.TektonV1beta1().PipelineRuns(namespace).Delete(context.TODO(), name, metav1.DeleteOptions{})
 }
 
+func (c *controller) cancelPipelineRun(pContext string, pr *pipelinev1beta1.PipelineRun) (*pipelinev1beta1.PipelineRun, error) {
+	p, err := c.getPipelineConfig(pContext)
+	if err != nil {
+		return nil, err
+	}
+	if pr.Spec.Status == pipelinev1beta1.PipelineRunSpecStatusCancelled {
+		return pr, nil
+	}
+	pr.Spec.Status = pipelinev1beta1.PipelineRunSpecStatusCancelled
+	return p.client.TektonV1beta1().PipelineRuns(pr.Namespace).Update(context.TODO(), pr, metav1.UpdateOptions{})
```

Comment:
I'd say that this should actually use a passed-in context instead of `context.TODO()`, otherwise how would we cancel?

Reply:
Oh, I see, the existing code already does not care... :unamused: I'd still like new code to do better, wdyt?

Reply:
This is on my list of fixes for the following PRs (with introducing a better interface for prowjob client, replacing the interface used here) so I would vote to leave it as it is and address it in the followup in the future.

Reply:
Yeah, that sounds good to me